configure with `--disable-builtin-atomics` cause compilation failure

<!--Thank you for taking the time to submit an issue!-->

## Background information

### What version of the PMIx Reference Library are you using? <!--(e.g., v1.0, v2.1, git master @ hash, etc.)-->

The one comes with openmpi main branch

```
commit e4c20e224e8ab32d1558ba273129e1d7b22b0836
Author: Samuel K. Gutierrez <samuel@lanl.gov>
Date:   Tue Apr 12 22:39:34 2022 -0600

    Fix memcpy size in pmix_bfrops_base_copy_darray().
    
    Fix incorrect memcpy size for the PMIX_PROC_RANK case found in
    pmix_bfrops_base_copy_darray().
    
    Signed-off-by: Samuel K. Gutierrez <samuel@lanl.gov>

```

### Describe how PMIx was installed <!--(e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)-->

compiled from source

### Please describe the system on which you are running

* Operating system/version: centos7
* Computer hardware: intel
* Network type: EFA

-----------------------------

## Details of the problem

to produce, do

```
./configure --disable-builtin-atomics
make
```
got the following error message:

```
...
make[2]: Leaving directory `/home/centos/PortaFiducia/libraries/openmpi/main/source/ompi/3rd-party/openpmix/src/mca/prm/slurm'
Making all in tools/pevent
make[2]: Entering directory `/home/centos/PortaFiducia/libraries/openmpi/main/source/ompi/3rd-party/openpmix/src/tools/pevent'
  CC       pevent.o
pevent.c: In function ‘evhandler_reg_callbk’:
pevent.c:87:5: warning: implicit declaration of function ‘pmix_atomic_wmb’ [-Wimplicit-function-declaration]
     PMIX_WAKEUP_THREAD(&lock->lock);
     ^
pevent.c: In function ‘main’:
pevent.c:234:5: warning: implicit declaration of function ‘pmix_atomic_rmb’ [-Wimplicit-function-declaration]
     PMIX_WAIT_THREAD(&mylock.lock);
     ^
  CCLD     pevent
pevent.o: In function `opcbfunc':
pevent.c:(.text+0x55): undefined reference to `pmix_atomic_wmb'
pevent.o: In function `evhandler_reg_callbk':
pevent.c:(.text+0xcf): undefined reference to `pmix_atomic_wmb'
pevent.o: In function `main':
pevent.c:(.text.startup+0x3ca): undefined reference to `pmix_atomic_rmb'
pevent.c:(.text.startup+0x56c): undefined reference to `pmix_atomic_rmb'
collect2: error: ld returned 1 exit status
make[2]: *** [pevent] Error 1
make[2]: Leaving directory `/home/centos/PortaFiducia/libraries/openmpi/main/source/ompi/3rd-party/openpmix/src/tools/pevent'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/centos/PortaFiducia/libraries/openmpi/main/source/ompi/3rd-party/openpmix/src'
make: *** [all-recursive] Error 1
```

--- a/config/pmix_config_asm.m4
+++ b/config/pmix_config_asm.m4
@@ -19,7 +19,7 @@ dnl Copyright (c) 2017-2021 Amazon.com, Inc. or its affiliates.  All Rights
 dnl                         reserved.
 dnl Copyright (c) 2020      Google, LLC. All rights reserved.
 dnl Copyright (c) 2020      Intel, Inc.  All rights reserved.
-dnl Copyright (c) 2021      Nanook Consulting.  All rights reserved.
+dnl Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
 dnl $COPYRIGHT$
 dnl
 dnl Additional copyrights may follow
@@ -102,6 +102,13 @@ AC_DEFUN([PMIX_CONFIG_ASM],[
         pmix_atomic_gcc_builtin=1
     elif test "$enable_builtin_atomics" = "yes" ; then
         AC_MSG_WARN([GCC built-in atomics requested but not found.])
+        AC_MSG_ERROR([Cannot continue])
+    else
+        AC_MSG_WARN([Neither C11 nor the built-in atomics are available,])
+        AC_MSG_WARN([either because they were disabled on the configure])
+        AC_MSG_WARN([command line or they were not found. PMIx requires])
+        AC_MSG_WARN([atomic support, so either a compiler with C11 atomics])
+        AC_MSG_WARN([must be used OR the built-in atomics must not be disabled.])
         AC_MSG_ERROR([Cannot continue])
     fi
 

